collector/database/tests/cltr_database_test.php

```diff
@@ -81,8 +81,14 @@ public function test_collector() {
     public function test_expiry() {
         global $DB;
 
-        $timeago = time() - (int)(19.5 * DAYSECS); // 19.5 days ago.
-        $stub = $this->get_metric_stub([1, 2, 3], $timeago, DAYSECS);
+        $tz = \core_date::get_server_timezone_object();
+
+        // Represents 10am, 20 days ago.
+        $datestr = '-20 days midnight +10 hours';
```

Comment:
Please also test 10PM, which will find its nearest midnight at 1200 not 0000

Reply:
Done